src/tokenserver/extractors.rs

```diff
@@ -102,13 +111,46 @@ impl FromRequest for TokenserverRequest {
                 hashed_fxa_uid,
                 hashed_device_id,
                 service_id,
+                duration: duration.unwrap_or(DEFAULT_TOKEN_DURATION),
             };
 
             Ok(tokenserver_request)
         })
     }
 }
 
+#[derive(Deserialize)]
+struct QueryParams {
+    pub duration: Option<String>,
```

Comment:
serde can make this conversion for us

```suggestion
    pub duration: Option<u64>,
```

Reply:
Won't the request fail if we let serde attempt the conversion to `u64` and the client sends a string? In that case, we just want to ignore the query param, but we don't want the request to fail altogether